ksiren/src/main/java/com/brightspace/ksiren/Action.kt

```diff
@@ -85,15 +106,24 @@ data class Action(
 			return obj
 		}
 	}
+}
 
-	fun hasField(name: String): Boolean {
-		this.fields.forEach {
-			if (it.name == name) {
-				return true
-			}
-		}
-		return false
-	}
+internal data class MutableAction(
+	override val name: String,
+	override val classes: List<String> = listOf(),
+	override val fields: List<MutableField> = listOf(),
+	override val href: String,
+	override val method: String = "GET",
+	override val title: String?,
+	override val type: ContentType = ContentType.FORM) : ActionBase {
 
-	override fun toJson() = JsonUtils.toJson(this)
+	constructor(action: ActionBase) : this(
+		action.name,
+		action.classes,
+		action.fields.map { MutableField(it) },
```

Comment:
Same as above, for clarity I'd suggest `action.fields.map({field -> MutableField(field)});